test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1

```diff
@@ -1,70 +1,98 @@
+            if ( (Get-ExperimentalFeature PSNativeCommandArgumentPassing).Enabled -and $PSNativeCommandArgumentPassing -ne "Legacy" ) {
+                $lines[0] | Should -BeExactly 'Arg 0 is <a\"b c\"d>'
+                $lines[1] | Should -BeExactly 'Arg 1 is <a\"b c\"d>'
+            }
+            else {
+                $lines[0] | Should -BeExactly 'Arg 0 is <a"b c"d>'
+                $lines[1] | Should -BeExactly 'Arg 1 is <a"b c"d>'
+            }
         }
-    }
 
-    It "Should handle PowerShell arrays with or without spaces correctly: <arguments>" -TestCases @(
-        @{arguments = "1,2"; expected = @("1,2")}
-        @{arguments = "1,2,3"; expected = @("1,2,3")}
-        @{arguments = "1, 2"; expected = "1,", "2"}
-        @{arguments = "1 ,2"; expected = "1", ",2"}
-        @{arguments = "1 , 2"; expected = "1", ",", "2"}
-        @{arguments = "1, 2,3"; expected = "1,", "2,3"}
-        @{arguments = "1 ,2,3"; expected = "1", ",2,3"}
-        @{arguments = "1 , 2,3"; expected = "1", ",", "2,3"}
-    ) {
-        param($arguments, $expected)
-        $lines = @(Invoke-Expression "testexe -echoargs $arguments")
-        $lines.Count | Should -Be $expected.Count
-        for ($i = 0; $i -lt $expected.Count; $i++) {
-            $lines[$i] | Should -BeExactly "Arg $i is <$($expected[$i])>"
+        It "Should correctly quote paths with spaces (ArgumentList=${PSNativeCommandArgumentPassing}): <arguments>" -TestCases @(
+            @{arguments = "'.\test 1\' `".\test 2\`""  ; expected = @(".\test 1\",".\test 2\")},
+            @{arguments = "'.\test 1\\\' `".\test 2\\`""; expected = @(".\test 1\\\",".\test 2\\")}
+        ) {
+            param($arguments, $expected)
+            $lines = Invoke-Expression "testexe -echoargs $arguments"
+            $lines.Count | Should -Be $expected.Count
+            for ($i = 0; $i -lt $lines.Count; $i++) {
+                $lines[$i] | Should -BeExactly "Arg $i is <$($expected[$i])>"
+            }
+        }
+
+        It "Should handle PowerShell arrays with or without spaces correctly (ArgumentList=${PSNativeCommandArgumentPassing}): <arguments>" -TestCases @(
+            @{arguments = "1,2"; expected = @("1,2")}
+            @{arguments = "1,2,3"; expected = @("1,2,3")}
+            @{arguments = "1, 2"; expected = "1,", "2"}
+            @{arguments = "1 ,2"; expected = "1", ",2"}
+            @{arguments = "1 , 2"; expected = "1", ",", "2"}
+            @{arguments = "1, 2,3"; expected = "1,", "2,3"}
+            @{arguments = "1 ,2,3"; expected = "1", ",2,3"}
+            @{arguments = "1 , 2,3"; expected = "1", ",", "2,3"}
+        ) {
+            param($arguments, $expected)
+            $lines = @(Invoke-Expression "testexe -echoargs $arguments")
+            $lines.Count | Should -Be $expected.Count
+            for ($i = 0; $i -lt $expected.Count; $i++) {
+                $lines[$i] | Should -BeExactly "Arg $i is <$($expected[$i])>"
+            }
         }
```

Comment:
Should have explicit test for equivalent to: `exe /c:"some value"`

Reply:
fixed